interop/src/clients/web.rs

```diff
@@ -0,0 +1,151 @@
+use std::net::SocketAddr;
+
+use color_eyre::eyre::Result;
+
+#[derive(Debug)]
+pub struct WebClient {
+    browser: fantoccini::Client,
+    client_id: Vec<u8>,
+}
+
+impl WebClient {
+    pub async fn new(driver_addr: &SocketAddr) -> Result<Self> {
+        let client_id = uuid::Uuid::new_v4();
+        let client_id_str = client_id.as_hyphenated().to_string();
+        let client_config = serde_json::json!({
+            "databaseName": format!("db-{client_id_str}"),
+            "key": "test",
+            "clientId": client_id_str
+        });
+        let browser = crate::build::web::webdriver::setup_browser(driver_addr).await?;
+
+        let _ = browser
+            .execute_async(
+                r#"
```

Comment:
It would be neat to have those in separate typescript files nearby and import them with `include_str!`. WDYT ?

Reply:
That's exactly what I used to do at the beginning, but I ditched the approach - it's a real PITA when you're refactoring and moving things around.
I think this could come back later if the JS becomes more complex. Right now it's just a couple of lines per "file", I don't think it's necessary